plugin_sge.py

```diff
@@ -163,103 +91,209 @@ def _extract_job_info(self, elem, elem_text, queue_name):
 
     def __repr__(self):
         return 'SGEStatMaker Instance'
 
 
-def _calc_everything(fn, write_method):
-    logging.debug('Parsing tree of %s' % fn)
-    anonymize = anonymize_func()
-    with open(fn, 'rb') as fin:
-        tree = etree.parse(fin)
-        root = tree.getroot()
-        worker_nodes = list()
-        existing_node_names = set()
-        # for queue_elem in root.iter('Queue-List'):  # 2.7-only
-        for queue_elem in root.findall('queue_info/Queue-List'):
-            worker_node = dict()
-            # worker_node['domainname'] = queue_elem.find('./resource[@name="hostname"]').text.split('.', 1)[0]  # 2.7 only
-            resources = queue_elem.findall('resource')
-            # TODO: find a way to loop ONCE for both hostname and qname!!
-            try:
-                slots_used = int(queue_elem.find('./slots_used').text)
-            except AttributeError:
-                slots_used = 0
-            count = 0
-            # worker_node.setdefault('qname', [])
-            for resource in resources:
-                if resource.attrib.get('name') == 'hostname':
-                    worker_node['domainname'] = resource.text if not options.ANONYMIZE else anonymize(resource.text, 'wns')
-                    count += 1
-                    if count == 2: break
-                elif resource.attrib.get('name') == 'qname':
-                    if not slots_used:
-                        worker_node['qname'] = set()
-                    else:
-                        worker_node['qname'] = set(resource.text[0]) \
-                            if not options.ANONYMIZE else set(anonymize(resource.text[0], 'qs'))
-                    count += 1
-                    if count == 2: break
-            else:
-                raise ValueError("No such resource")
+class SGEBatchSystem(object):
+    def __init__(self, in_out_filenames, config):
+        self.sge_file_stat = in_out_filenames.get('sge_file_stat')
+        self.sge_file_stat_out = in_out_filenames.get('sge_file_stat_out')
+        # self.temp_filepath = SGEStatMaker.temp_filepath
 
-            # worker_node['np'] = queue_elem.find('./resource[@name="num_proc"]').text  # python 2.7 only
-            resources = queue_elem.findall('resource')
-            for resource in resources:
-                if resource.attrib.get('name') == 'num_proc':
-                    worker_node['np'] = resource.text
-                    break
-            else:
-                # TODO: check this for bugs, maybe raise an exception in the future?
-                worker_node['np'] = 0
+        self.config = config
+        self.sge_stat_maker = SGEStatMaker(self.config)
+
+    def convert_inputs(self):
+        return self._serialise_qstat()
+
+    def get_queues_info(self):
+        return self._get_statq_from_xml(self.sge_file_stat)
 
+    def get_worker_nodes(self):
+        return self._calc_everything(self.sge_file_stat)
+
+    @staticmethod
+    def get_jobs_info():
```

Comment:
I don't get this. The specific get_jobs_info only uses a file that exists in SGEStatMaker. I *could* turn it into a normal method, but why? what are the consquences ?